CA-195752: Disable core parking by default

Core parking is used by firmware to offline CPUs for reasons such as
thermal and power management. However, offlining CPUs has exposed bugs
in other areas (the scheduler). Disable core parking by default to
prevent these issues, but add a command-line switch ("core_parking") to
re-enable if needed.

diff --git a/xen/arch/x86/platform_hypercall.c b/xen/arch/x86/platform_hypercall.c
index b9d05f8416..b84a29cdf0 100644
--- a/xen/arch/x86/platform_hypercall.c
+++ b/xen/arch/x86/platform_hypercall.c
@@ -9,6 +9,7 @@
 #include <xen/types.h>
 #include <xen/lib.h>
 #include <xen/mm.h>
+#include <xen/param.h>
 #include <xen/sched.h>
 #include <xen/domain.h>
 #include <xen/event.h>
@@ -46,6 +47,9 @@ struct resource_access {
     xenpf_resource_entry_t *entries;
 };
 
+static bool __read_mostly opt_core_parking_enabled;
+boolean_param("core_parking", opt_core_parking_enabled);
+
 long cf_check cpu_frequency_change_helper(void *);
 void check_resource_access(struct resource_access *);
 void cf_check resource_access(void *);
@@ -732,6 +736,12 @@ ret_t do_platform_op(
         switch(op->u.core_parking.type)
         {
         case XEN_CORE_PARKING_SET:
+            if ( !opt_core_parking_enabled )
+            {
+                ret = -EINVAL;
+                goto out;
+            }
+
             idle_nums = min_t(uint32_t,
                     op->u.core_parking.idle_nums, num_present_cpus() - 1);
             if ( CONFIG_NR_CPUS > 1 )
